File: kotlin-eclipse-ui/src/org/jetbrains/kotlin/ui/editors/StringStorage.java
package org.jetbrains.kotlin.ui.editors;

import java.io.ByteArrayInputStream;
import java.io.InputStream;

import org.eclipse.core.resources.IStorage;
import org.eclipse.core.runtime.CoreException;
import org.eclipse.core.runtime.IPath;
import org.jetbrains.annotations.NotNull;

public class StringStorage implements IStorage {
    
    private final @NotNull String content;
    private final @NotNull String name;
    private final @NotNull String packageFqName;
    
    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        result = prime * result + name.hashCode();
        result = prime * result + packageFqName.hashCode();
        result = prime * result + content.hashCode();
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        StringStorage other = (StringStorage) obj;
        return name.equals(other.name) &&
                packageFqName.equals(other.packageFqName) && 
                content.equals(other.content);
    }
    
    StringStorage(@NotNull String input, @NotNull String name, @NotNull String packageFqName) {
      this.content = input;
      this.name = name;
      this.packageFqName = packageFqName;
    }
   
    @Override
    public InputStream getContents() throws CoreException {
      return new ByteArrayInputStream(content.getBytes());
    }
   
    @Override
    public IPath getFullPath() {
      return null;
    }
   
    @Override
    public <T> T getAdapter(Class<T> adapter) {
        return null;
    }
    
    @Override
    public String getName() {
        return name;
    }
   
    @Override
    public boolean isReadOnly() {
      return true;
    }
    
    public String getFqName() {
        return packageFqName + '/' + name;
    }
}

File: kotlin-eclipse-ui/src/org/jetbrains/kotlin/ui/editors/navigation/StringStorage.kt
/*******************************************************************************
 * Copyright 2000-2016 JetBrains s.r.o.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 *******************************************************************************/
package org.jetbrains.kotlin.ui.editors.navigation

import org.eclipse.core.resources.IStorage
import org.eclipse.core.runtime.IPath
import java.io.InputStream

data class StringStorage(
        private val content: String,
        private val name: String,
        private val packageFqName: String) : IStorage {
    
    override fun getName(): String = name

    override fun getContents(): InputStream? = content.byteInputStream()

    override fun getFullPath(): IPath? = null

    override fun <T> getAdapter(adapter: Class<T>?): T? = null

    override fun isReadOnly(): Boolean = true

    val fqName: String
        get() = "$packageFqName/$name"
}